parsons/databases/mysql/mysql.py

```diff
@@ -0,0 +1,188 @@
+        cur = connection.cursor(buffered=True)
+
+        try:
+            yield cur
+        finally:
+            cur.close()
+
+    def query(self, sql, parameters=None):
+        """
+        Execute a query against the database. Will return ``None``if the query returns zero rows.
+
+        To include python variables in your query, it is recommended to pass them as parameters,
+        following the `mysql style <https://security.openstack.org/guidelines/dg_parameterize-database-queries.html>`_.
+        Using the ``parameters`` argument ensures that values are escaped properly, and avoids SQL
+        injection attacks.
+
+        **Parameter Examples**
+
+        .. code-block:: python
+
+            # Note that the name contains a quote, which could break your query if not escaped
+            # properly.
+            name = "Beatrice O'Brady"
+            sql = "SELECT * FROM my_table WHERE name = %s"
+            mysql.query(sql, parameters=[name])
+
+        .. code-block:: python
+
+            names = ["Allen Smith", "Beatrice O'Brady", "Cathy Thompson"]
+            placeholders = ', '.join('%s' for item in names)
+            sql = f"SELECT * FROM my_table WHERE name IN ({placeholders})"
+            mysql.query(sql, parameters=names)
+
+        `Args:`
+            sql: str
+                A valid SQL statement
+            parameters: list
+                A list of python variables to be converted into SQL values in your query
+
+        `Returns:`
+            Parsons Table
+                See :ref:`parsons-table` for output options.
+
+        """  # noqa: E501
+
+        with self.connection() as connection:
+            return self.query_with_connection(sql, connection, parameters=parameters)
+
+    def query_with_connection(self, sql, connection, parameters=None, commit=True):
+        """
+        Execute a query against the database, with an existing connection. Useful for batching
+        queries together. Will return ``None`` if the query returns zero rows.
+
+        `Args:`
+            sql: str
+                A valid SQL statement
+            connection: obj
+                A connection object obtained from ``mysql.connection()``
+            parameters: list
+                A list of python variables to be converted into SQL values in your query
+            commit: boolean
+                Whether to commit the transaction immediately. If ``False`` the transaction will
+                be committed when the connection goes out of scope and is closed (or you can
+                commit manually with ``connection.commit()``).
+
+        `Returns:`
+            Parsons Table
+                See :ref:`parsons-table` for output options.
+        """
+        with self.cursor(connection) as cursor:
+
+            # The python connector can only execute a single sql statement, so we will
+            # break up each statement and execute them separately.
+            for s in sql.strip().split(';'):
```

Comment:
It looks like there's a `multi` keyword which can be passed into the `execute` call. [From what I can tell](https://overiq.com/mysql-connector-python-101/executing-queries-using-connector-python/#executing-multiple-queries-with-execute), it looks like the return values is a list of result sets.. not sure how that works with the `fetchmany` method below.

I definitely think we should lean on the connector for parsing out the different statements.. if I have a query that contains a `;` in a string (e.g, `select * from tmc_data.table where response = 'a;b;c'`), then `split` above will create 3 strings, none of which is the complete statement. Not a super likely scenario, but given enough users, someone will probably hit it.